legion/engine/llri-vk/utils.cpp

```diff
@@ -0,0 +1,119 @@
+#include <llri/llri.hpp>
+#include <vulkan/vulkan.hpp>
+#include <map>
+#include <string>
+#include <vector>
+
+namespace legion::graphics::llri
+{
+    namespace internal
+    {
+        /**
+         * @brief Helper function that maps layers to their names
+        */
+        std::map<std::string, vk::LayerProperties> preprocessLayerProperties()
+        {
+            std::map<std::string, vk::LayerProperties> result;
+
+            std::vector<vk::LayerProperties> availableLayers = vk::enumerateInstanceLayerProperties();
+            for (auto layer : availableLayers)
+                result[layer.layerName.data()] = layer;
```

Comment:
using `emplace` here is faster than `[]` you almost never want to use square brackets unless you don't know what items already exist in the map and you want to read an item even if it doesn't exist. for purely writing, just use `emplace` or `try_emplace`.